firebase-common/src/main/java/com/google/firebase/internal/HeartBeatInfoStorage.java

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.internal;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import androidx.annotation.RestrictTo;
+import androidx.annotation.VisibleForTesting;
+
+/**
+ * Class responsible for storing all heartbeat related information. Store a key value pair with the
+ * key being either the heartbeat tag (one for each firebase sdk) or the global heart beat tag and
+ * the value being the timestamp(in millis) as to when the last heart beat was sent. Class also
+ * provides helper functions to check if there is a need to send the global/sdk heart beat.
```

Comment:
I can't see any helper functions, also any private implementation details don't need to be documented.